localstack/aws/gateway.py

```diff
@@ -34,3 +35,15 @@ def process(self, request: Request, response: Response):
         context.request = request
 
         chain.handle(context, response)
+
+    def accept(self, request: WebSocketRequest):
+        response = Response(status=101)
+        self.process(request, response)
+
+        # only send the populated response if the websocket hasn't already done so before
+        if response.status_code != 101:
+            if request.is_upgraded():
+                return
+            if request.is_rejected():
+                return
+            request.reject(response)
```

Comment:
nit: So this basically covers the case when a routed endpoint does not call accept on the request but sets a response, right?
Maybe a small test for this would be great as well, just to make sure that this control flow is covered.